libraries/networking/src/PacketReceiver.cpp

```diff
@@ -190,6 +218,132 @@ void PacketReceiver::unregisterListener(QObject* listener) {
     _directConnectSetMutex.unlock();
 }
 
+void PacketReceiver::handleVerifiedPacketList(std::unique_ptr<udt::PacketList> packetList) {
+    // if we're supposed to drop this packet then break out here
+    if (_shouldDropPackets) {
+        return;
+    }
+
+    // setup an NLPacketList from the PacketList we were passed
+    auto nlPacketList = new NLPacketList(std::move(*packetList));
+
+    auto nodeList = DependencyManager::get<LimitedNodeList>();
+    
+    _inPacketCount += nlPacketList->getNumPackets();
+    _inByteCount += nlPacketList->getDataSize();
+    
+    SharedNodePointer matchingNode;
+    
+    if (!nlPacketList->getSourceID().isNull()) {
+        matchingNode = nodeList->nodeWithUUID(nlPacketList->getSourceID());
+        
+        if (matchingNode) {
+            // No matter if this packet is handled or not, we update the timestamp for the last time we heard
+            // from this sending node
+            matchingNode->setLastHeardMicrostamp(usecTimestampNow());
+        }
+    }
+    
+    _packetListenerLock.lock();
```

Comment:
Do we have any "best practices" on this? Should we use a scoped locker even if it's protecting one line? I'm inclined to say "always use scoped locking" - the benefits seem to outweigh the small costs.